Indicators/BetaIndicator.cs

```diff
@@ -0,0 +1,171 @@
+        /// <param name="targetSymbol">The target symbol of this indicator</param>
+        /// <param name="referenceSymbol">The reference symbol of this indicator</param>
+        public BetaIndicator(string name, int period, Symbol targetSymbol, Symbol referenceSymbol)
+            : base(name)
+        {
+            // Assert the period is greater than two, otherwise the beta can not be computed
+            if (period < 2)
+            {
+                throw new Exception($"Period parameter for Beta indicator must be greater than 2 but was {period}");
+            }
+
+            WarmUpPeriod = period;
+            _referenceSymbol = referenceSymbol;
+            _targetSymbol = targetSymbol;
+
+            _symbolDataPoints = new Dictionary<Symbol, RollingWindow<decimal>>();
+            _symbolDataPoints.Add(_referenceSymbol, new RollingWindow<decimal>(WarmUpPeriod));
+            _symbolDataPoints.Add(_targetSymbol, new RollingWindow<decimal>(WarmUpPeriod));
+
+            _targetReturns = new RollingWindow<double>(WarmUpPeriod);
+            _referenceReturns = new RollingWindow<double>(WarmUpPeriod);
+            _beta = 0;
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// 
+        /// As this indicator is receiving data points from two different symbols,
+        /// it's going to compute the next value when the amount of data points
+        /// of each of them is the same. Otherwise, it will return the last beta
+        /// value computed
+        /// </summary>
+        /// <param name="input">The input value of this indicator on this time step.
+        /// It can be either from the target or the reference symbol</param>
+        /// <returns>The beta value of the target used in relation with the reference</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            var inputSymbol = input.Symbol;
+            _symbolDataPoints[inputSymbol].Add(input.Close);
+
+            if (_symbolDataPoints[_targetSymbol].Samples == _symbolDataPoints[_referenceSymbol].Samples && _symbolDataPoints[_targetSymbol].Samples > 1)
+            {
+                _targetReturns.Add(GetNewReturn(_symbolDataPoints[_targetSymbol]));
+                _referenceReturns.Add(GetNewReturn(_symbolDataPoints[_referenceSymbol]));
+
+                ComputeBeta();
+            }
+            return _beta;
+        }
+
+        /// <summary>
+        /// Computes the returns with the new given data point and the last given data point
+        /// </summary>
+        /// <param name="rollingWindow">The collection of data points from which we want
+        /// to compute the return</param>
+        /// <returns>The returns with the new given data point</returns>
+        private static double GetNewReturn(RollingWindow<decimal> rollingWindow)
+        {
+            return (double) ((rollingWindow[1] / rollingWindow[0]) - 1);
```

Comment:
Hm you have a rolling window of size `WarmUpPeriod` but only really using the last 2 values in it?